fms/datasets/__init__.py

```diff
@@ -0,0 +1,20 @@
+from fms.datasets.instructions import JsonInstructions
+from fms.datasets import text
+from fms.utils.tokenizers import BaseTokenizer
+
+__dataset_factory = {"instruction": JsonInstructions, "text": text.causaltext}
+
+
+def get_dataset(name: str, tokenizer: BaseTokenizer, data: str = None):
+    """
+    Get a dataset by type.
+
+    Args:
+    name: Which style of dataset to use. E.g. instruction, text.
+    data: The data this dataset should load. E.g. a url or file system path.
+    """
+    name = name.lower()
+    if name not in __dataset_factory:
+        raise NameError(f"Dataset name should be one of {__dataset_factory.keys()}")
+    ctr = __dataset_factory[name]
+    return ctr(data, tokenizer)
```

Comment:
I see JsonInstructions and  text.causaltext takes more input, device, max_len, ignore_index, etc. Do we want to handle that?

Reply:
I think we can add them later. The immediate goal of this PR is to make it easy to write a (tuning) script that uses these datasets, and I'm probably not going to bother exposing all of those options as command-line flags in the tuning script. I think other args can be added easily though as-needed / if-needed later.